src/streamsync/ui_manager.py

```diff
@@ -115,9 +117,57 @@ def _prepare_value(value):
     return str(value)
 
 
-def _create_component(component_tree: ComponentTree,  component_type: str, **kwargs) -> Component:
+def _check_parent_child_relations(
+        component_tree: ComponentTree,
+        parent_id: str,
+        component_type: str
+):
+    # Import required inside a function:
+    # StreamsyncUIManager class stores actual type maps,
+    # but importing it directly causes a circular import
+    from streamsync.ui import StreamsyncUIManager
+
+    if component_type == "root":
+        raise UIError("Root component cannot be a child component.")
+
+    parent = component_tree.get_component(parent_id)
+    if not parent:
+        raise RuntimeError(
+            f"Improper parent_id provided: {parent_id} is missing in tree"
+            )
+
+    valid_children_types_for_parent = \
+        StreamsyncUIManager.children_types_map.get(parent.type)
+    valid_parent_types_for_component = \
+        StreamsyncUIManager.parent_types_map.get(component_type)
+
+    if not valid_children_types_for_parent \
+       or not valid_parent_types_for_component:
+        type_to_blame = (f"Parent type '{parent.type}'"
+                         if not valid_children_types_for_parent
+                         else f"Child type '{component_type}'")
+        raise RuntimeError(
+            f"Misconfigured types: {type_to_blame} is not present " +
+            "in allowed types map."
+            )
+
+    is_component_eligible = \
```

Comment:
Are we checking "inherit"?

Reply:
No, my apologies, I wasn't aware of it. Can you explain it a bit so I can account for it?

Reply:
Ah, I think I understand now. If we're talking `Repeater`, we need to check on its parent for valid types.

Reply:
You have the frontend logic here https://github.com/streamsync-cloud/streamsync/blob/dev/ui/src/core/typeHierarchy.ts

For example, if you have a _Column Container_, which inside has a _Repeater_, which in turn has another _Repeater_ inside, the 2nd (nested) _Repeater_ should only allow _Column_, _Repeater_ and I believe _Separator_.

It makes things harder. I'm not sure if it's worth replicating this in the backend. I'd do it in the frontend, which would also helps us with `ui.json` manipulation/corruption.